File: columbu/CATscala/src/main/java/com/columbu/p2p/net/peer/PeerConnectionDelegate.java
package com.galileo.p2p.net.peer;

import com.galileo.core.overlay.message.Message;
import com.galileo.utils.Sha256Hash;
import com.galileo.p2p.net.message.TronMessage;

public abstract class PeerConnectionDelegate {

  public abstract void onMessage(PeerConnection peer, TronMessage msg);

  public abstract Message getMessage(Sha256Hash msgId);

  public abstract void onConnectPeer(PeerConnection peer);

  public abstract void onDisconnectPeer(PeerConnection peer);

}

File: columbu/CATscala/src/main/scala/com/columbu/p2p/net/peer/PeerConnectionDelegate.scala
package com.galileo.p2p.net.peer

import com.galileo.p2p.message.Message
import com.galileo.utils.Sha256Hash
import com.galileo.p2p.net.message.TronMessage

abstract class PeerConnectionDelegate {
  def onMessage(peer: PeerConnection, msg: TronMessage)

  def getMessage(msgId: Sha256Hash): Message

  def onConnectPeer(peer: PeerConnection)

  def onDisconnectPeer(peer: PeerConnection)

}
